File: wheystore/wheystore.Web/Modules/SanPhamDB/DanhMucSanPham/DanhMucSanPhamColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.DanhMucSanPham")]
[BasedOnRow(typeof(DanhMucSanPhamRow), CheckNames = true)]
public class DanhMucSanPhamColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MaLoai { get; set; }
    [EditLink]
    public string TenLoai { get; set; }
    public string MoTa { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/DanhMucSanPhamColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DanhMucSanPhamRow } from "./DanhMucSanPhamRow";

export interface DanhMucSanPhamColumns {
    MaLoai: Column<DanhMucSanPhamRow>;
    TenLoai: Column<DanhMucSanPhamRow>;
    MoTa: Column<DanhMucSanPhamRow>;
}

export class DanhMucSanPhamColumns extends ColumnsBase<DanhMucSanPhamRow> {
    static readonly columnsKey = 'SanPhamDB.DanhMucSanPham';
    static readonly Fields = fieldsProxy<DanhMucSanPhamColumns>();
}